engine/rpcserver.go

```diff
@@ -1832,39 +1889,290 @@ func (s *RPCServer) GCTScriptListAll(context.Context, *gctrpc.GCTScriptListAllRe
 	}
 
 	if r.Status {
 		err := gctscript.Autoload(r.Script, true)
 		if err != nil {
-			return &gctrpc.GCTScriptGenericResponse{Status: "error", Data: err.Error()}, nil
+			return &gctrpc.GenericResponse{Status: "error", Data: err.Error()}, nil
 		}
-		return &gctrpc.GCTScriptGenericResponse{Status: "success", Data: "script " + r.Script + " removed from autoload list"}, nil
+		return &gctrpc.GenericResponse{Status: "success", Data: "script " + r.Script + " removed from autoload list"}, nil
 	}
 
 	err := gctscript.Autoload(r.Script, false)
 	if err != nil {
-		return &gctrpc.GCTScriptGenericResponse{Status: "error", Data: err.Error()}, nil
+		return &gctrpc.GenericResponse{Status: "error", Data: err.Error()}, nil
+	}
+	return &gctrpc.GenericResponse{Status: "success", Data: "script " + r.Script + " added to autoload list"}, nil
+}
+
+// SetExchangeAsset enables or disables an exchanges asset type
+func (s *RPCServer) SetExchangeAsset(_ context.Context, r *gctrpc.SetExchangeAssetRequest) (*gctrpc.GenericSubsystemResponse, error) {
+	exch := GetExchangeByName(r.Exchange)
+	if exch == nil {
+		return nil, errExchangeNotLoaded
+	}
+
+	exchCfg, err := Bot.Config.GetExchangeConfig(r.Exchange)
+	if err != nil {
+		return nil, err
+	}
+
+	base := exch.GetBase()
+	if base == nil {
+		return nil, errExchangeBaseNotFound
+	}
+
+	if r.Asset == "" {
+		return nil, errors.New("asset type must be specified")
+	}
+
+	a := asset.Item(r.Asset)
+	err = base.CurrencyPairs.SetAssetEnabled(a, r.Enable)
+	if err != nil {
+		return nil, err
+	}
+	err = exchCfg.CurrencyPairs.SetAssetEnabled(a, r.Enable)
+	if err != nil {
+		return nil, err
+	}
+
+	return &gctrpc.GenericSubsystemResponse{}, nil
+}
+
+// SetAllExchangePairs enables or disables an exchanges pairs
+func (s *RPCServer) SetAllExchangePairs(_ context.Context, r *gctrpc.SetExchangeAllPairsRequest) (*gctrpc.GenericSubsystemResponse, error) {
+	exch := GetExchangeByName(r.Exchange)
+	if exch == nil {
+		return nil, errExchangeNotLoaded
+	}
+
+	exchCfg, err := Bot.Config.GetExchangeConfig(r.Exchange)
+	if err != nil {
+		return nil, err
+	}
+
+	base := exch.GetBase()
+	if base == nil {
+		return nil, errExchangeBaseNotFound
+	}
+
+	assets := base.CurrencyPairs.GetAssetTypes()
+
+	if r.Enable {
+		for i := range assets {
+			var pairs currency.Pairs
+			pairs, err = base.CurrencyPairs.GetPairs(assets[i], false)
+			if err != nil {
+				return nil, err
+			}
+			exchCfg.CurrencyPairs.StorePairs(assets[i], pairs, true)
+			base.CurrencyPairs.StorePairs(assets[i], pairs, true)
+		}
+	} else {
+		for i := range assets {
+			exchCfg.CurrencyPairs.StorePairs(assets[i], nil, true)
+			base.CurrencyPairs.StorePairs(assets[i], nil, true)
+		}
+	}
+
+	if exch.IsWebsocketEnabled() && base.Websocket.IsConnected() {
+		err = exch.FlushWebsocketChannels()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &gctrpc.GenericSubsystemResponse{}, nil
```

Comment:
Can we please add a response here even if its "true"?
Just to let the user know it was successful